pageserver/src/page_service.rs

```diff
@@ -2195,12 +2195,16 @@ impl PageServerHandler {
         if request_lsn < **latest_gc_cutoff_lsn && !timeline.is_gc_blocked_by_lsn_lease_deadline() {
             let gc_info = &timeline.gc_info.read().unwrap();
             if !gc_info.lsn_covered_by_lease(request_lsn) {
-                return Err(
-                    PageStreamError::BadRequest(format!(
-                        "tried to request a page version that was garbage collected. requested at {} gc cutoff {}",
-                        request_lsn, **latest_gc_cutoff_lsn
-                    ).into())
+                // While request was in flight, replica apply_lsn may be advanced.
+                // latest_gc_cutoff_lsn is conservative estimation for min(redo_lsn) for all replicas,
+                // so it is safe to move request_lsn forward to latest_gc_cutoff_lsn.
+                // If replica lease is expired and latest_gc_cutoff_lsn>redo_lsn for this replica,
+                // then check of page LSN at replia protects it from getting too new version of the page.
```

Comment:
This PR only adds the check for RO replicas in the communicator, but, this method here is also executed for requests from primary and static computes.
Practically, 
- primary sends request_lsn=0xfff..., so, never going to hit the outer `if` conditino, but, that's not obvious to future readers of this code and
- static compute sends request_lsn=$static_lsn , which, may very well hit this branch here and raise the request_lsn

I guess in contrast to commentary in communicator.c, the check there also applies to static computes?
But, not super obvious, at very least the comment there requires adjustment.

Reply:
If LSN requested by static replica is collected by GC and we have to advance `request_lsn` to `gc_cutoff` here, then it will be detected by check in compute that returned page LSN is greater than `redo_lsn=static_lsn`.